backtrace-library/src/main/java/backtraceio/library/breadcrumbs/BacktraceBreadcrumbsLogger.java

```diff
@@ -0,0 +1,88 @@
+package backtraceio.library.breadcrumbs;
+
+import java.io.IOException;
+import java.util.Map;
+
+import backtraceio.library.enums.BacktraceBreadcrumbLevel;
+import backtraceio.library.enums.BacktraceBreadcrumbType;
+import backtraceio.library.logger.BacktraceLogger;
+
+public class BacktraceBreadcrumbsLogger {
+
+    private final String LOG_TAG = BacktraceBreadcrumbsLogger.class.getSimpleName();
+
+    private long breadcrumbId = 0;
+
+    private BacktraceQueueFileHelper backtraceQueueFileHelper;
+
+    /**
+     * We truncate messages longer than this
+     */
+    private final int maxMessageSizeBytes = 1024;
+
+    /**
+     * We truncate serialized attribute strings longer than this
+     */
+    private final int maxSerializedAttributeSizeBytes = 1024;
+
+    public BacktraceBreadcrumbsLogger(String breadcrumbLogDirectory, int maxQueueFileSizeBytes) throws IOException, NoSuchMethodException {
+        this.backtraceQueueFileHelper = new BacktraceQueueFileHelper(breadcrumbLogDirectory, maxQueueFileSizeBytes);
+    }
+
+    public boolean addBreadcrumb(String message, Map<String, Object> attributes, BacktraceBreadcrumbType type, BacktraceBreadcrumbLevel level) {
+        // We use currentTimeMillis in the BacktraceReport too, so for consistency
+        // we will use it here.
+        long time = System.currentTimeMillis();
+
+        message = message.substring(0, Math.min(message.length(), maxMessageSizeBytes));
+
+        StringBuilder breadcrumb = new StringBuilder("\ntimestamp " + Long.toString(time));
+        breadcrumb.append(" id ");
+        breadcrumb.append(breadcrumbId);
+        breadcrumb.append(" level ");
+        breadcrumb.append(level.toString());
+        breadcrumb.append(" type ");
+        breadcrumb.append(type.toString());
+        breadcrumb.append(" attributes ");
+        breadcrumb.append(serializeAttributes(attributes));
+        breadcrumb.append(" message ");
+        breadcrumb.append(message.replace("\n", ""));
+        breadcrumb.append("\n");
+
+        breadcrumbId++;
+
+        return backtraceQueueFileHelper.add(breadcrumb.toString().getBytes());
+    }
+
+    public boolean clear() {
+        return backtraceQueueFileHelper.clear();
+    }
+
+    private String serializeAttributes(final Map<String, Object> attributes) {
+        StringBuilder serializedAttributes = new StringBuilder();
+        if (attributes != null) {
+            for (Map.Entry<String, Object> entry : attributes.entrySet()) {
```

Comment:
I'm not a big fun of this proposal :( If we anyway create our own protocol, maybe we can store attributes in bytes instead of custom string parameters? Or maybe we should convert attributes to JSON object?  Or maybe escape some values like `\n` with `\\n` ?

Reply:
We can convert the entire message string to JSON format. We just won't be able to make the whole file a valid JSON (maintaining brackets), but each individual message string can be JSON.

Reply:
I proposed this to @amadden80. I think it could make things easier for both of us.